cdf_fabric_replicator/data_modeling.py

```diff
@@ -227,6 +218,30 @@ def write_instance_to_lakehouse(
         self.state_store.set_state(external_id=state_id, high=json.dumps(res.cursors))
         self.state_store.synchronize()
 
+    def sync_instances(self, query):
+        try:
+            res = self.cognite_client.data_modeling.instances.sync(query=query)
+        except CogniteAPIError:
+            query.cursors = None  # type: ignore
+            try:
+                res = self.cognite_client.data_modeling.instances.sync(query=query)
+            except CogniteAPIError as e:
+                self.logger.error(f"Failed to sync instances. Error: {e}")
+                raise e
+        return res
+
+    def has_recent_update(self, result: QueryResult, query_start_time: int) -> bool:
+        # Check if any node/edge has lastUpdatedTime >= query_start_time
+        if "nodes" in result.data:
+            for node in result.data["nodes"]:
+                if node.get("lastUpdatedTime", 0) >= query_start_time:
+                    return True
+            if "edges" in result.data:
+                for edge in result.data["edges"]:
```

Comment:
Is the indent here one too much? In this case, if you only have edges, you will not look at edges.